language/move-prover/src/prelude.bpl

```diff
@@ -131,16 +131,111 @@ function {:inline} $IsValidNum(v: $Value): bool {
 }
 
 
+
+{{#if backend.vector_using_sequences}}
+
+
+// This is the implementation of $ValueArray using sequences
+// Value Array
+// -----------
+
+type {:datatype} {:builtin "(Seq T@$Value)"} $ValueArray;
+
+function {:builtin "(as seq.empty (Seq T@$Value))"} $EmptyValueArray(): $ValueArray;
+
+function {:builtin "seq.nth"} $ReadValueArray(a: $ValueArray, i: int): $Value;
+
+function {:builtin "seq.len"} $LenValueArray(a: $ValueArray): int;
+
+function {:builtin "seq.extract"} $ValueArrayExtract(a: $ValueArray, offset: int, length: int): $ValueArray;
+
+function {:builtin "seq.++"} $ConcatValueArray(a: $ValueArray, b:$ValueArray): $ValueArray;
+
+function {:builtin "seq.unit"} $UnitValueArray(v: $Value): $ValueArray;
+
+
+function {{backend.func_inline}} $RemoveValueArray(a: $ValueArray): $ValueArray {
+    (
+        $ValueArrayExtract(a, 0, $LenValueArray(a) -1)
+    )
+}
+
+
+function {{backend.func_inline}} $RemoveIndexValueArray(a: $ValueArray, i: int): $ValueArray {
+    (
+        var prefix := $ValueArrayExtract(a, 0, i); (
+        var suffix := $ValueArrayExtract(a, i+1, $LenValueArray(a)-i-1);
+        $ConcatValueArray(prefix, suffix)
+    ))
+}
+
+function {{backend.func_inline}} $ReverseValueArray(a: $ValueArray): $ValueArray;
+axiom (forall a: $ValueArray :: $LenValueArray($ReverseValueArray(a)) == $LenValueArray(a));
+axiom (forall a: $ValueArray :: (forall i: int :: $ValueArrayExtract(a, i, 1) == $ValueArrayExtract(a, $LenValueArray(a)-1-i, 1)));
+// The second axiom works outside the legal range because in z3 seq.extract outside the legal range degenerates to the empty sequence
+// Not using seq.nth because of weird behaviour outside the legal range
```

Comment:
Because you commenting practice is not consistent it is unclear what definition you are referring too. In most software languages the convention is:

[empty line]
// comment about the thing
definition of the thing